src/averageAges.js

```diff
@@ -37,7 +47,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothers = people
+    .map((human) => human.mother)
+    .filter(name => name !== null);
+
+  const forWomen = people
+    .filter((human) => human.sex === 'f')
+    .filter((human) => !withChildren || mothers.includes(human.name));
```

Comment:
the same. you can usw just one filter instead of 3 filters and 1 map. To check child you can use find method